modules/clightning/module.cpp

```diff
@@ -279,6 +280,54 @@ std::optional<std::string> clightning_parse_p2p_lightning_message(std::span<cons
 
         result << "MSG_TYPE=warning;CHANNEL_ID=" << hex_encode(channel.id, 32);
         result << ";DATA=" << tal_hex(tmpctx, data);
+    } else if (msg_type == WIRE_INIT) {
+        u8 *globalfeatures, *features;
+        struct tlv_init_tlvs *tlvs;
+        struct wireaddr *remote_addr = nullptr;
+
+        if (!fromwire_init(tmpctx, msg, &globalfeatures, &features, &tlvs)) {
+            return "";
+        }
+
+        if (tlvs->remote_addr) {
+            const u8 *cursor = tlvs->remote_addr;
+            size_t len = tal_bytelen(tlvs->remote_addr);
+
+            remote_addr = tal(tmpctx, struct wireaddr);
+            if (!fromwire_wireaddr(&cursor, &len, remote_addr)) {
+                return "";
+            }
+            // C-lightning accepts extra trailing bytes in address parsing and
+            // rust-lightning doesn't. So we skip.
+            if (len != 0) {
+                return std::nullopt;
+            }
+        }
+
+        u8 *features_combined = featurebits_or(tmpctx, globalfeatures, features);
+
+        result << "MSG_TYPE=init";
+        result << ";FEATURES=" << tal_hex(tmpctx, features_combined);
+
+        if (tlvs->networks) {
+            result << ";NETWORKS=";
+            for (size_t i = 0; i < tal_count(tlvs->networks); ++i) {
+                result << hex_encode(tlvs->networks[i].shad.sha.u.u8, 32) << ";";
```

Comment:
Since all the networks apply to the `NETWORKS=`, it would be more readable to separate them with `,` instead of `;`.
```suggestion
                result << hex_encode(tlvs->networks[i].shad.sha.u.u8, 32) << ",";
```

Reply:
Updated!